Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_Apple.cpp

```diff
@@ -80,13 +80,37 @@ ST::string pfApplePasswordStore::GetPassword(const ST::string& username)
 bool pfApplePasswordStore::SetPassword(const ST::string& username, const ST::string& password)
 {
     ST::string service = GetServerDisplayName();
+    
+    OSStatus err = SecKeychainAddGenericPassword(nullptr,
+                                                service.size(),
+                                                service.c_str(),
+                                                username.size(),
+                                                username.c_str(),
+                                                password.size(),
+                                                password.c_str(),
+                                                nullptr);
+    if(err==errSecDuplicateItem) {
+        //the keychain item already exists, update it
+        CFStringRef cfUsername = CFStringCreateWithCString(NULL, username.c_str(), kCFStringEncodingUTF8);
+        CFDataRef cfPassword = CFDataCreate(NULL, (const UInt8*)password.c_str(), password.size());
+        CFStringRef cfServiceName = CFStringCreateWithCString(NULL, service.c_str(), kCFStringEncodingUTF8);
```

Comment:
```suggestion
        CFStringRef cfUsername = CFStringCreateWithCString(nullptr, username.c_str(), kCFStringEncodingUTF8);
        CFDataRef cfPassword = CFDataCreate(nullptr, (const UInt8*)password.c_str(), password.size());
        CFStringRef cfServiceName = CFStringCreateWithCString(nullptr, service.c_str(), kCFStringEncodingUTF8);
```

We prefer `nullptr` for strongly typed null or the integer 0 when a pointer is inappropriate.

Reply:
Making these changes. I think this is a good change for cpp files. In Obj-C things are going to get a little weird. Obj-C prefers NULL, and I think nullptr is only available in Obj-C++. Usually NULL is used with CF types but I think in since this is a C++ file it's better just to hold CF to the C++ standards. Functionally CF doesn't care anyway.